tests/cpp17/test_cpp17.cpp

```diff
@@ -38,6 +39,138 @@ namespace cpp11 {
 #include "../optional_scalars_generated.h"
 }  // namespace cpp11
 
+using ::cpp17::MyGame::Example::Monster;
+using ::cpp17::MyGame::Example::Vec3;
+
+/*******************************************************************************
+** Build some FB objects.
+*******************************************************************************/
+Vec3 BuildVec3() {
+  using ::cpp17::MyGame::Example::Color;
+  using ::cpp17::MyGame::Example::Test;
+  return Vec3(
+    /*x=*/1.1,
+    /*y=*/2.2,
+    /*z=*/3.3,
+    /*test1=*/6.6,
+    /*test2=*/Color::Green,
+    /*test3=*/Test(
+      /*a=*/11,
+      /*b=*/90
+    )
+  );
+}
+
+const Monster* BuildMonster(flatbuffers::FlatBufferBuilder& fbb) {
+  using ::cpp17::MyGame::Example::MonsterBuilder;
```

Comment:
why do we need another function creating a buffer? don't we have existing buffers we can test?

Reply:
I wanted to use a Monster object for this test because it has many fields of all different types (including nested structs) in order to exercise all of the possibilities.  However, in this particular test file (test_cpp27.cpp) there is not an existing function that creates a Monster object like this, so I have added this one.  If there is another one somewhere else that you are aware of that I can reuse (which has all the different field types represented and populated) let me know and I am happy to reuse it.

Reply:
`CreateMonsterDirect` + `GetBuffer` (or `GetBufferSpan`) + `flatbuffers::GetRoot<Monster>`?

Reply:
I started changing to this approach (`CreateMonsterDirect`) but that seems a bit inconvenient in this case because it will force me to put many parameters corresponding to fields that I am not using, since I am populating a few fields towards the end, I cannot rely on the default values and I would have many parameters just `nullptr` or `0`.  Perhaps in this case where we are just populating a few select fields the Builder approach is better?